app/src/main/java/com/github/polybooks/activities/SaleInformationActivity.kt

```diff
@@ -20,31 +27,61 @@ class SaleInformationActivity: AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_sale_information)
 
-        val viewTitle: TextView = findViewById(R.id.sale_information_title)
-        val viewEdition: TextView = findViewById(R.id.sale_information_edition)
-        val viewAuthors: TextView = findViewById(R.id.sale_information_authors)
+        val sale = (intent.getSerializableExtra(EXTRA_SALE_INFORMATION) as Sale)
+
+        findViewById<TextView>(R.id.sale_information_title).text = sale.book.title
+        findViewById<TextView>(R.id.sale_information_edition).text = sale.book.edition
+        findViewById<TextView>(R.id.sale_information_authors).text = StringsManip.listAuthorsToString(sale.book.authors)
         // val countryFlag: TextView = findViewById(R.id.countryFlag)
-        val viewPublishDate: TextView = findViewById(R.id.sale_information_book_publish_date)
-        val viewPublisher: TextView = findViewById(R.id.sale_information_book_publisher)
-        val viewFormat: TextView = findViewById(R.id.sale_information_book_format)
+        findViewById<TextView>(R.id.sale_information_book_format).text = sale.book.format
         // val bookImage = findViewById(R.id.proof_picture)
 
-        val viewCondition: TextView = findViewById(R.id.sale_information_condition)
-        val viewPrice: TextView = findViewById(R.id.sale_information_price)
+        val ratingBar: RatingBar = findViewById(R.id.sale_information_rating)
+        ratingBar.rating = 0f
 
+        if (sale.book.totalStars != null && sale.book.numberVotes != null && sale.book.numberVotes != 0) {
+            ratingBar.rating = (sale.book.totalStars / sale.book.numberVotes).toFloat()
+        }
+        ratingBar.setOnRatingBarChangeListener { bar, rating, _ ->
+            val firestore = FirebaseFirestore.getInstance()
+            val olBookDB = OLBookDatabase { string -> url2json(string) }
```

Comment:
Same as before, you should never create the databases yourself, or pick a specific implementation, but use `Database.bookDatabase`